lib/archethic/self_repair/resync.ex

```diff
@@ -0,0 +1,55 @@
+defmodule Archethic.SelfRepair.Resync do
+  @moduledoc """
+   Handle Resync Capabilites of Protocol Chains.
+  """
+
+  alias Archethic.{Crypto, P2P, TransactionChain, Replication, SelfRepair}
+
+  require Logger
+
+  @spec node_out_of_sync() :: :ok
+  def node_out_of_sync() do
+    first_address = Crypto.genesis_address(:node_shared_secrets)
+
+    case SelfRepair.repair_in_progress?(first_address) do
+      false ->
+        SelfRepair.start_worker(
+          first_address: first_address,
+          storage_address: first_address,
```

Comment:
Here the RepairWorker expect the storage address to be the transaction to replicate. It will not fetch the last transaction of the chain and replicate it. So here you should provide the transaction you want to replicate (in fact the last nss transaction).
To achieve this, maybe the function `P2P.is_synced_node?()` could be renamed to `P2P.verify_synchronization` and this funtion could return either `:ok` if it is synchronized or `{:error, last_nss_address}`. So the right address to synchronize could be passed to the `Resync.node_out_of_sync` function.